### PR TITLE
added proxy download and fixed the other download test

### LINKED ISSUE #178: Add proxy download testing
We currently don't have any automated testing for proxy.

I believe this should be doable via our proxy helper we use the agent and serving up a little server to serve up the local built artifact. Worst case, could probably allow it to to go all the way out to the download site... that just gets funky on mac cause there are no pre-builts.

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added integration tests for downloading with proxy
 * Fixed the download tests to skip when running on branch or unsupported architectures

## Links
Closes #178 

## Details
I can't seem to get a programatic proxy server to work when calling pre-build with the cli which is why i'm using `executeCli`.  I could change the proxy server to be run on the command line as well

### CODE REVIEW

File: lib/pre-build.js [hunk 1]
```diff
@@ -212,13 +212,14 @@ function download(target, cb) {
   let client = null
   let hasCalledBack = false
   let options = {}
+  const proxyHost = process.env.NR_NATIVE_METRICS_PROXY_HOST
 
-  if (process.env.NR_NATIVE_METRICS_PROXY_HOST) {
-    const proxy = process.env.NR_NATIVE_METRICS_PROXY_HOST
-    options.host = new URL(DOWNLOAD_HOST).host
+  if (proxyHost) {
+    const parsedUrl = new URL(DOWNLOAD_HOST)
```
Comment (bizob2828) [inline lib/pre-build.js:218]: I uncovered this bug while writing tests.  The `NR_NATIVE_METRICS_DOWNLOAD_HOST` can only be a hostname and not specify a port.  This prob only affects us but makes it impossible to specify an alternate host + port.

File: tests/integration/pre-build.tap.js [hunk 2]
```diff
@@ -47,17 +72,42 @@ tap.test('pre-build commands', function (t) {
...
+    execSync(`node ./lib/pre-build --no-build install native_metrics`)
+    const binary = findBinary()
+    t.match(
+      binary,
+      /_newrelic_native_metrics-\d{1,3}_\d{1,3}_\d{1,3}-native_metrics.*\.node/,
+      'should download binary'
+    )
+    proxyServer.kill()
+    downloadServer.kill()
   })
```
Comment (michaelgoin) [inline tests/integration/pre-build.tap.js:111]: Wonder if we could add a way to confirm we went through the proxy to the test.

**Review by michaelgoin — APPROVED** [review]
💯  🎉 🥇

File: tests/integration/pre-build.tap.js (hunk 2 above)
Comment (bizob2828) [reply tests/integration/pre-build.tap.js:111]: Hmm, not sure. I'll think on it

**Pull request merged**